Terminal.Gui/Core/Responder.cs

```diff
@@ -277,6 +277,9 @@ protected virtual void Dispose (bool disposing)
 
 				// TODO: free unmanaged resources (unmanaged objects) and override finalizer
 				// TODO: set large fields to null
+#if DEBUG_IDISPOSABLE
+				Instances.Remove(this);
+#endif
```

Comment:
This is on the wrong place. Also this doesn't guarantee that all instances were really disposed (`WasDisposed is true`).
I'll send you a PR to your branch which will include the fix for this and for the unit tests failure.

Reply:
Here is the fix https://github.com/a-usr/Terminal.Gui-FORK/pull/1.

Reply:
> This is on the wrong place. Also this doesn't guarantee that all instances were really disposed (`WasDisposed is true`). I'll send you a PR to your branch which will include the fix for this and for the unit tests failure.

however it does remove the current instance from the instances static, doesn't it? 
Compared to checking the entire static every time one element is being disposed, this would also have better performance?

Anyways, i have merged your Pull Request.

Reply:
@a-usr, sorry I forgot to do that. I already sent to your branch a fix for this on https://github.com/a-usr/Terminal.Gui-FORK/pull/4.
This really increase significantly the performance. Thanks for pointing this.